spark/src/main/java/org/opensearch/sql/spark/execution/statestore/StateStore.java

```diff
@@ -255,64 +255,83 @@ public static Function<AsyncQueryJobMetadata, AsyncQueryJobMetadata> createJobMe
+                        XContentCommonAttributes.TYPE,
+                        SessionModelXContentSerializer.SESSION_DOC_TYPE))
                 .must(
                     QueryBuilders.termQuery(
-                        SessionModel.SESSION_STATE, SessionState.RUNNING.getSessionState())));
-  }
-
-  public static BiFunction<FlintIndexStateModel, FlintIndexState, FlintIndexStateModel>
-      updateFlintIndexState(StateStore stateStore, String datasourceName) {
-    return (old, state) ->
-        stateStore.updateState(
-            old,
-            state,
-            FlintIndexStateModel::copyWithState,
-            DATASOURCE_TO_REQUEST_INDEX.apply(datasourceName));
+                        SessionModelXContentSerializer.SESSION_TYPE,
+                        SessionType.INTERACTIVE.getSessionType()))
+                .must(
+                    QueryBuilders.termQuery(
+                        XContentCommonAttributes.STATE, SessionState.RUNNING.getSessionState())));
   }
 
   public static Supplier<Long> activeRefreshJobCount(StateStore stateStore, String datasourceName) {
     return () ->
         stateStore.count(
-            DATASOURCE_TO_REQUEST_INDEX.apply(datasourceName),
+            OpenSearchStateStoreUtil.getIndexName(datasourceName),
             QueryBuilders.boolQuery()
                 .must(
                     QueryBuilders.termQuery(
-                        SessionModel.TYPE, FlintIndexStateModel.FLINT_INDEX_DOC_TYPE))
-                .must(QueryBuilders.termQuery(STATE, FlintIndexState.REFRESHING.getState())));
+                        XContentCommonAttributes.TYPE,
+                        FlintIndexStateModelXContentSerializer.FLINT_INDEX_DOC_TYPE))
+                .must(
+                    QueryBuilders.termQuery(
+                        XContentCommonAttributes.STATE, FlintIndexState.REFRESHING.getState())));
   }
 
   public static Supplier<Long> activeStatementsCount(StateStore stateStore, String datasourceName) {
     return () ->
         stateStore.count(
-            DATASOURCE_TO_REQUEST_INDEX.apply(datasourceName),
+            OpenSearchStateStoreUtil.getIndexName(datasourceName),
             QueryBuilders.boolQuery()
                 .must(
-                    QueryBuilders.termQuery(StatementModel.TYPE, StatementModel.STATEMENT_DOC_TYPE))
+                    QueryBuilders.termQuery(
+                        XContentCommonAttributes.TYPE,
+                        StatementModelXContentSerializer.STATEMENT_DOC_TYPE))
                 .should(
                     QueryBuilders.termsQuery(
-                        StatementModel.STATEMENT_STATE,
+                        XContentCommonAttributes.STATE,
                         StatementState.RUNNING.getState(),
                         StatementState.WAITING.getState())));
   }
+
+  @SuppressWarnings("unchecked")
+  private <T extends StateModel> XContentSerializer<T> getXContentSerializer(T st) {
```

Comment:
XContentSerializer still depend on OpenSearch library?

Reply:
I'm not quite sure the intention of the question, but XContentSerializer is more on OpenSearch side since the XContent is OpenSearch specific content serialization. It won't be used for non-opensearch environment.